File: modules/stego-core/src/main/java/com/shterneregen/securelan/stego/model/StegoContentType.java
package com.shterneregen.securelan.stego.model;

import java.util.Arrays;

public enum StegoContentType {
    BINARY((byte) 1),
    UTF8_TEXT((byte) 2);

    private final byte code;

    StegoContentType(byte code) {
        this.code = code;
    }

    public byte code() {
        return code;
    }

    public static StegoContentType fromCode(byte code) {
        return Arrays.stream(values())
                .filter(type -> type.code == code)
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("Unsupported steganography content type: " + code));
    }
}

File: modules/stego-core/src/main/kotlin/com/shterneregen/securelan/stego/model/StegoContentType.kt
package com.shterneregen.securelan.stego.model

enum class StegoContentType(private val codeValue: Byte) {
    BINARY(1),
    UTF8_TEXT(2);

    fun code(): Byte = codeValue

    companion object {
        @JvmStatic
        fun fromCode(code: Byte): StegoContentType = entries
            .firstOrNull { it.codeValue == code }
            ?: throw IllegalArgumentException("Unsupported steganography content type: $code")
    }
}
